pkg/api/customization/kontainerdriver/validator.go

```diff
@@ -0,0 +1,43 @@
+package kontainerdriver
+
+import (
+	"fmt"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type Validator struct {
+	KontainerDriverLister v3.KontainerDriverLister
+}
+
+func (v *Validator) Validator(request *types.APIContext, schema *types.Schema, data map[string]interface{}) error {
+	var spec v3.KontainerDriverSpec
+
+	if err := convert.ToObj(data, &spec); err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, "Kontainer driver spec conversion error")
+	}
+
+	if err := v.validateKontainerDriverURL(request, spec); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
You can move the validate check to the end of the func like
```
return v.validateKontainerDriverURL(request, spec)
```
There is no need for an error check.